c7n/resources/s3.py

```diff
@@ -375,6 +377,314 @@ def get_permissions(self):
         return self.permissions
 
 
+class LifecycleApiMixin(object):
+    """Mixin class for calling the s3 lifecycle API in a multuthreaded fashion."""
```

Comment:
this mixin can go away, we just annotate buckets with lifecycle attributes when we assemble the bucket subdocuments in the resource manager (the s3 augment table), i believe its already there.